runtime/src/main/java/org/corfudb/protocols/wireprotocol/CorfuMsgType.java

```diff
@@ -61,9 +61,10 @@
     FILL_HOLE(34, new TypeToken<CorfuPayloadMsg<TrimRequest>>() {}),
     PREFIX_TRIM(38, new TypeToken<CorfuPayloadMsg<TrimRequest>>() {}),
     TAIL_REQUEST(41, TypeToken.of(CorfuMsg.class), true),
-    TAIL_RESPONSE(42, new TypeToken<CorfuPayloadMsg<Long>>(){}, true),
+    TAIL_RESPONSE(42, new TypeToken<CorfuPayloadMsg<TailsResponse>>(){}, true),
     COMPACT_REQUEST(43, TypeToken.of(CorfuMsg.class), true),
     FLUSH_CACHE(44, TypeToken.of(CorfuMsg.class), true),
+    // why do many of these operations bypass the epoch check?
```

Comment:
a github issue would be a better place to keep track of this concern

Reply:
done.